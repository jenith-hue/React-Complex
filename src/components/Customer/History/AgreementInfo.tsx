import {
  makeStyles,
  RACCOLOR,
  RACTable,
  RACTableCell,
  RACTableRow,
  Typography,
} from '@rentacenter/racstrap';
import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import InfiniteScroll from 'react-infinite-scroll-component';
import {
  ApiStateWrapper,
  Loader,
} from '../../../common/ApiStateWrapper/ApiStateWrapper';
import {
  useAgreementInfo,
  useAgreementInfoActions,
} from '../../../context/AgreementInfo/AgreementInfoProvider';
import { formatDateString } from '../../../utils/utils';
import { AgreementInfoDaysLate } from '../../../types/types';
import { AppRoute } from '../../../config/route-config';
import { useCustomerDetails } from '../../../context/CustomerDetails/CustomerDetailsProvider';

const useClasses = makeStyles((theme: any) => ({
  tableContentColor: {
    color: RACCOLOR.INDEPENDENCE,
  },
  tableRoot: {
    marginTop: theme.typography.pxToRem(16),
  },
  // aligns the content in the cell
  storeRowContentCell: {
    display: 'flex',
  },
  storeRowContent: {
    position: 'absolute',
  },
  notVisible: {
    visibility: 'hidden',
  },
  tableBody: {
    position: 'relative',
  },
  tableHead: {
    position: 'sticky',
    top: 0,
    zIndex: 2,
    background: 'white',
  },
  contentHeight: {
    height: theme.typography.pxToRem(400),
  },
  greenColumn: {
    backgroundColor: '#81c783',
  },
  number: {
    color: RACCOLOR.BLUE_CRAYOLA,
    cursor: 'pointer',
  },
}));

export const EMPTY_DAYS_LATE_HISTORY = [
  { dayslate: 'On_Time', count: 0 },
  { dayslate: '1-6_Days_Late', count: 0 },
  { dayslate: '7-14_Days_Late', count: 0 },
  { dayslate: '15+_Days_Late', count: 0 },
];

export const populateDaysLateHistory = (
  daysLateHistory: AgreementInfoDaysLate[] | null | undefined
): AgreementInfoDaysLate[] => {
  return EMPTY_DAYS_LATE_HISTORY.map((emptyDaysLateObj) => {
    const daysLateObjFromAPI = daysLateHistory?.find(
      (obj) => obj.dayslate === emptyDaysLateObj.dayslate
    );

    return daysLateObjFromAPI || emptyDaysLateObj;
  });
};

export const AgreementInfo = () => {
  const { stores, total, numberOfLoadedAgreements, hasApiError, loading } =
    useAgreementInfo();
  const { customerDetails } = useCustomerDetails();
  const { customerId } = customerDetails || {};

  const renderTableHead = () => (
    <>
      <RACTableCell>
        <Typography variant="h5">Agreement</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Due Date</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Type</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Current</Typography>
      </RACTableCell>
      <RACTableCell color="green">
        <Typography className={classes.greenColumn} variant="h5">
          On Time
        </Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">1-6 Days Late</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">7-14 Days Late</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">15+ Days Late</Typography>
      </RACTableCell>
    </>
  );

  const daysLateHistoryCellGenerator = (
    daysLateHistory: AgreementInfoDaysLate[]
  ) => {
    const populatedDaysLateHistory = populateDaysLateHistory(daysLateHistory);

    return populatedDaysLateHistory.map(
      ({ count, dayslate }: AgreementInfoDaysLate) => (
        <RACTableCell key={dayslate}>
          <Typography variant="body2" className={classes.tableContentColor}>
            {count}
          </Typography>
        </RACTableCell>
      )
    );
  };
  const renderTableContent = () => (
    <>
      {stores?.map((storeObj) => {
        if (!storeObj?.agreements?.length) return null;

        return storeObj?.agreements?.map(
          (
            {
              agreementNumber,
              agreementId,
              nextDueDate,
              currentDaysLate,
              daysLateHistory,
              contractType,
            },
            index
          ) => {
            const dueDateString = `${formatDateString(nextDueDate)}`;
            const showAgreementNumLink = contractType === 'CLUB' ? false : true;
            return (
              <>
                {index === 0 && (
                  <RACTableRow
                    key={`${index}-${storeObj.storeNumber}`}
                    backgroundColor="white"
                  >
                    <RACTableCell className={classes.storeRowContentCell}>
                      <div className={classes.storeRowContent}>
                        <Typography variant="h4">{`${storeObj.storeNumber} - ${storeObj.city}, ${storeObj.state}`}</Typography>
                      </div>
                    </RACTableCell>
                    <RACTableCell>
                      <Typography
                        noWrap
                        display="inline"
                        variant="body2"
                        className={classes.notVisible}
                      >
                        {/* Using typography to ensure that all rows have the same
                        height(this is not visible in the table)*/}
                        0
                      </Typography>
                    </RACTableCell>
                  </RACTableRow>
                )}
                <RACTableRow key={index} backgroundColor="white">
                  <RACTableCell classes={{ root: classes.number }}>
                    {showAgreementNumLink ? (
                      <Link
                        style={{ color: 'inherit', textDecoration: 'unset' }}
                        to={{
                          pathname: `${AppRoute.CustomerAggrement}/${customerId}/${agreementId}`,
                          search: `?origin=am-customer`,
                        }}
                        className={'disi'}
                      >
                        {agreementNumber}
                      </Link>
                    ) : (
                      <Typography
                        display="inline"
                        variant="body2"
                        className={classes.tableContentColor}
                      >
                        {agreementNumber}
                      </Typography>
                    )}
                  </RACTableCell>
                  <RACTableCell>
                    <Typography
                      display="inline"
                      variant="body2"
                      className={classes.tableContentColor}
                    >
                      {dueDateString}
                    </Typography>
                  </RACTableCell>
                  <RACTableCell>
                    <Typography
                      display="inline"
                      variant="body2"
                      className={classes.tableContentColor}
                    >
                      {contractType}
                    </Typography>
                  </RACTableCell>
                  <RACTableCell>
                    <Typography
                      display="inline"
                      variant="body2"
                      className={classes.tableContentColor}
                    >
                      {currentDaysLate}
                    </Typography>
                  </RACTableCell>
                  {daysLateHistoryCellGenerator(daysLateHistory)}
                </RACTableRow>
              </>
            );
          }
        );
      })}
    </>
  );
  const classes = useClasses();
  const { fetchAgreementInfo } = useAgreementInfoActions();

  const initialLoad = () => {
    fetchAgreementInfo();
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(initialLoad, []);
  return (
    <InfiniteScroll
      next={fetchAgreementInfo}
      dataLength={numberOfLoadedAgreements}
      hasMore={numberOfLoadedAgreements < total}
      height={330}
      loader={<Loader />}
    >
      <ApiStateWrapper
        loading={loading}
        hasApiError={hasApiError}
        response={stores}
        successContent={
          <RACTable
            bodyClasses={{ root: classes.tableBody }}
            headClasses={{ root: classes.tableHead }}
            classes={{ root: classes.tableRoot }}
            renderTableHead={renderTableHead}
            renderTableContent={renderTableContent}
          />
        }
        classes={{
          loader: classes.contentHeight,
          apiError: classes.contentHeight,
          noItems: classes.contentHeight,
        }}
      />
    </InfiniteScroll>
  );
};
