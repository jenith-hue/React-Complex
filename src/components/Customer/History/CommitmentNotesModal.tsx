import {
  RACModal,
  RACButton,
  makeStyles,
  RACCOLOR,
  TextField,
  RACDatePicker,
  RACTimePicker,
  Typography,
  RACTextField,
} from '@rentacenter/racstrap';

import React from 'react';
import { WorkedHistoryCommitmentType } from '../../../types/types';
import { useCustomerPaymentSummary } from '../../../context/CustomerPaymentSummary/CustomerPaymentSummaryProvider';
import clsx from 'clsx';
import { Commitment } from '../../../domain/PastDueList/PastDueCustomerList';

interface ModalProps {
  commitment: WorkedHistoryCommitmentType | Commitment;
  onClose: () => void;
}
const useStyles = makeStyles((theme: any) => ({
  dialogContent: {
    textAlign: 'left',
    height: theme.typography.pxToRem(400),
  },
  dialogRoot: {
    '& .MuiDialogContent-root': {
      padding: '1rem',
    },
    '& .MuiDialog-paperWidthSm': {
      maxWidth: theme.typography.pxToRem(800),
      maxHeight: theme.typography.pxToRem(407),
    },
    '& .MuiTypography-h4': {
      fontSize: '20px !important',
    },
    '& .MuiOutlinedInput-multiline': {
      padding: theme.typography.pxToRem(10),
    },
    '& .MuiInputBase-multiline': {
      margin: '8px 0px 0px 0px !important',
    },
    '& .MuiInputBase-input': {
      fontFamily: 'OpenSans-regular',
      fontSize: theme.typography.pxToRem(14),
    },
  },
  notesLabel: {
    color: theme.palette.text.primary,
    transform: 'scale(1) !important',
    ...theme.typography.body1,
    position: 'relative',
    display: 'block',
  },
  amountLabel: {
    color: theme.palette.text.primary,
    transform: 'scale(1) !important',
    ...theme.typography.body1,
  },
  dialogActions: {
    paddingRight: theme.typography.pxToRem(15),
    paddingBottom: theme.typography.pxToRem(15),
  },
  notesWrapper: {
    display: 'inline-flex',
    position: 'relative',
    flexDirection: 'column',
    width: '100%',
    marginTop: theme.typography.pxToRem(24),
    marginBotton: theme.typography.pxToRem(16),
  },
  modalColumn: {
    width: '25%',
    paddingRight: theme.typography.pxToRem(12),
  },
  row: {
    display: 'flex',
  },
  topBottomLeftBorderRadius: {
    borderTopLeftRadius: `${theme.typography.pxToRem(9)} !important`,
    borderBottomLeftRadius: `${theme.typography.pxToRem(9)} !important`,
    borderBottomRightRadius: 0,
    borderTopRightRadius: 0,
  },
  topBottomRightBorderRadius: {
    borderTopLeftRadius: 0,
    borderBottomLeftRadius: 0,
    borderBottomRightRadius: `${theme.typography.pxToRem(9)} !important`,
    borderTopRightRadius: `${theme.typography.pxToRem(9)} !important`,
  },
  amountValue: {
    padding: `${theme.typography.pxToRem(6)} ${theme.typography.pxToRem(
      12
    )} ${theme.typography.pxToRem(6)} 0`,
    backgroundColor: RACCOLOR.ICEWIND_DALE,
    // Replace with SATIN_WHITE
    border: `${theme.typography.pxToRem(1)} solid ${RACCOLOR.ICEWIND_DALE}`,
    lineHeight: theme.typography.pxToRem(18),
    width: '90%',
    marginLeft: theme.typography.pxToRem(-1),
  },
  amountWrapper: {
    display: 'flex',
    flexDirection: 'column',
  },
  amountValueWrapper: {
    marginTop: theme.typography.pxToRem(3),
    width: '100%',
    display: 'inline-flex',
  },
  amount$Value: {
    width: '10%',
    padding: `${theme.typography.pxToRem(6)} ${theme.typography.pxToRem(12)}`,
    backgroundColor: RACCOLOR.ICEWIND_DALE,
    // Replace with SATIN_WHITE
    border: `${theme.typography.pxToRem(1)} solid ${RACCOLOR.ICEWIND_DALE}`,
    lineHeight: theme.typography.pxToRem(18),
  },
}));

interface CommitmentNotesModalContentProps {
  commitment: Commitment;
}

const CommitmentNotesModalContent = ({
  commitment,
}: CommitmentNotesModalContentProps) => {
  const classes = useStyles();
  const { customerPaymentSummary: customerPaymentSummaryResponse } =
    useCustomerPaymentSummary();
  const {
    notes,
    commitmentDate,
    commitmentType,
    commitmentNotes,
    commitmentAmount,
  } = commitment || {};
  const splitCommitmentDate = commitmentDate?.split('T');
  const amount =
    customerPaymentSummaryResponse?.pastDueSummary?.amountDue ||
    commitmentAmount ||
    '';

  return (
    <>
      <div className={classes.row}>
        <div className={classes.modalColumn}>
          <RACTextField
            name="CommitmentType"
            label="Commitment Type"
            value={commitmentType?.description || ''}
            type="text"
            disabled
          />
        </div>
        <div className={classes.modalColumn}>
          <RACDatePicker
            label="Date"
            disabled
            value={splitCommitmentDate?.[0] || ''}
            name="commitmentDate"
          />
        </div>
        <div className={classes.modalColumn}>
          <RACTimePicker
            disabled
            label="Time"
            name="commitmentTime"
            value={splitCommitmentDate?.[1] || ''}
          />
        </div>
        <div className={classes.modalColumn}>
          <label className={classes.amountLabel}>Amount</label>
          <div className={classes.amountValueWrapper}>
            <Typography
              display="inline"
              variant="body2"
              className={clsx(
                classes.amount$Value,
                classes.topBottomLeftBorderRadius
              )}
            >
              $
            </Typography>
            <Typography
              display="inline"
              variant="body2"
              className={clsx(
                classes.amountValue,
                classes.topBottomRightBorderRadius
              )}
            >
              {amount}
            </Typography>
          </div>
        </div>
      </div>
      <div className={classes.notesWrapper}>
        <label className={classes.notesLabel}>Notes</label>
        <TextField
          multiline
          variant="outlined"
          disabled
          value={notes || commitmentNotes || ''}
          key="logActivityNote"
          maxRows={4}
          minRows={47}
        />
      </div>
    </>
  );
};

export const CommitmentNotesModal = ({ commitment, onClose }: ModalProps) => {
  const classes = useStyles();

  return (
    <RACModal
      isOpen
      titleVariant="h4"
      classes={{
        dialogContent: classes.dialogContent,
        dialog: classes.dialogRoot,
        dialogActions: classes.dialogActions,
      }}
      maxWidth="sm"
      title="Commitment Notes"
      content={<CommitmentNotesModalContent commitment={commitment} />}
      onClose={onClose}
      buttons={
        <RACButton variant="contained" color="primary" onClick={onClose}>
          Ok
        </RACButton>
      }
    />
  );
};
