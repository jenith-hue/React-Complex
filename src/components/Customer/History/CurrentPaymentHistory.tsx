import {
  makeStyles,
  RACButton,
  RACCOLOR,
  RACTable,
  RACTableCell,
  RACTableRow,
  Typography,
} from '@rentacenter/racstrap';
import React, { useEffect, useState } from 'react';
import InfiniteScroll from 'react-infinite-scroll-component';
import { useLocation } from 'react-router-dom';
import {
  ApiStateWrapper,
  Loader,
} from '../../../common/ApiStateWrapper/ApiStateWrapper';
import { days, RECEIPT_TYPE } from '../../../constants/constants';
import {
  usePaymentHistory,
  usePaymentHistoryActions,
} from '../../../context/PaymentHistory/PaymentHistoryProvider';
import { CustomerLocationState } from '../../../types/types';
import {
  formatDateString,
  formatStringDateHoursAndMinutes,
} from '../../../utils/utils';
import { Document } from '../../Document/Document';

const useClasses = makeStyles((theme: any) => ({
  tableContentColor: {
    color: RACCOLOR.INDEPENDENCE,
  },
  tableRoot: {
    marginTop: theme.typography.pxToRem(16),
  },
  tableBody: {
    position: 'relative',
  },
  tableHead: {
    position: 'sticky',
    top: 0,
    zIndex: 2,
    background: 'white',
  },
  contentHeight: {
    height: theme.typography.pxToRem(400),
  },
}));

export const CurrentPaymentHistory = () => {
  const [receiptId, setReceiptId] = useState('');

  const {
    payments,
    total,
    numberOfLoadedPayments,
    hasApiError,
    loading,
    init,
  } = usePaymentHistory();

  const renderTableHead = () => (
    <>
      <RACTableCell>
        <Typography variant="h5">Date paid</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Day</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Time</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Receipt #</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Total Payment</Typography>
      </RACTableCell>
      <RACTableCell align="center">
        <Typography variant="h5">Next Due Date</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Days Ext</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5"># Days late</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Store</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Payment Origin</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Tender Type</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Net Rent</Typography>
      </RACTableCell>
      <RACTableCell>
        <Typography variant="h5">Club</Typography>
      </RACTableCell>
    </>
  );

  const renderTableContent = () => (
    <>
      {payments?.map((dataObj, index) => {
        const paymentDate = new Date(dataObj.paymentDate || '');
        return (
          <RACTableRow key={index} hover backgroundColor="white">
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {formatDateString(dataObj.paymentDate)}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {days[paymentDate.getDay()]}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {formatStringDateHoursAndMinutes(dataObj.paymentDate)}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <RACButton
                variant="text"
                size="small"
                color="primary"
                onClick={() => {
                  // eslint-disable-next-line no-console
                  console.log('Implement mee');
                }}
              >
                <Typography
                  variant="body2"
                  color="primary"
                  onClick={() => {
                    setReceiptId(dataObj.receiptId as string);
                  }}
                >
                  {dataObj.receiptId}
                </Typography>
              </RACButton>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.paymentAmount !== 'null'
                  ? `$${dataObj.paymentAmount}`
                  : '-'}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography
                align="center"
                variant="body2"
                className={classes.tableContentColor}
              >
                {formatDateString(dataObj.nextDueDate)}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.daysExtension}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.daysLate}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.storeNumber}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj?.paymentOrigin?.description || ''}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.paymentMethodType?.join(', ')}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.netRentalRevenue !== 'null'
                  ? `$${dataObj.netRentalRevenue}`
                  : '-'}
              </Typography>
            </RACTableCell>
            <RACTableCell>
              <Typography variant="body2" className={classes.tableContentColor}>
                {dataObj.clubPayment !== 'null'
                  ? `$${dataObj.clubPayment}`
                  : '-'}
              </Typography>
            </RACTableCell>
          </RACTableRow>
        );
      })}
    </>
  );
  const classes = useClasses();
  const { fetchPaymentHistory, resetPaymentData } = usePaymentHistoryActions();

  const initialLoad = () => {
    fetchPaymentHistory(true);
    return resetPaymentData;
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(initialLoad, []);

  const location = useLocation<CustomerLocationState>();
  useEffect(() => {
    if (!location?.state?.customer) return;
    initialLoad();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [location?.state?.customer]);

  return (
    <>
      {receiptId && (
        <Document
          title="Receipt"
          documentId={receiptId}
          typeId={RECEIPT_TYPE}
          onClose={() => setReceiptId('')}
        />
      )}
      <InfiniteScroll
        next={fetchPaymentHistory}
        dataLength={numberOfLoadedPayments}
        hasMore={numberOfLoadedPayments < total}
        height={330}
        loader={<Loader />}
      >
        <ApiStateWrapper
          loading={loading && init}
          hasApiError={hasApiError}
          response={payments}
          successContent={
            <RACTable
              bodyClasses={{ root: classes.tableBody }}
              headClasses={{ root: classes.tableHead }}
              classes={{ root: classes.tableRoot }}
              renderTableHead={renderTableHead}
              renderTableContent={renderTableContent}
            />
          }
          classes={{
            loader: classes.contentHeight,
            apiError: classes.contentHeight,
            noItems: classes.contentHeight,
          }}
        />
      </InfiniteScroll>
    </>
  );
};
