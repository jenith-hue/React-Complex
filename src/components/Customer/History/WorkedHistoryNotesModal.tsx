import {
  makeStyles,
  RACButton,
  RACCOLOR,
  RACModal,
  Typography,
} from '@rentacenter/racstrap';
import React from 'react';

export interface ModalProps {
  workedResultText: string;
  noteText: string;
  open: boolean;
  onClose: () => void;
}
export const useStyles = makeStyles((theme: any) => ({
  dialogContent: {
    textAlign: 'left',
  },
  dialogRoot: {
    '& .MuiDialogContent-root': {
      padding: '1rem',
      flexDirection: 'column',
      display: 'flex',
      height: 'unset',
      gap: theme.typography.pxToRem(20),
    },
    '& .MuiDialog-paperWidthSm': {
      maxWidth: theme.typography.pxToRem(500),
      maxHeight: theme.typography.pxToRem(363),
    },
    '& .MuiTypography-h4': {
      fontSize: '20px !important',
    },
    '& .MuiOutlinedInput-multiline': {
      padding: theme.typography.pxToRem(10),
    },
    '& .MuiInputBase-multiline': {
      margin: '8px 0px 0px 0px !important',
    },
    '& .MuiInputBase-input': {
      fontFamily: 'OpenSans-regular',
      fontSize: theme.typography.pxToRem(14),
    },
  },
  dialogActions: {
    paddingRight: theme.typography.pxToRem(15),
    paddingBottom: theme.typography.pxToRem(15),
    justifyContent: 'center',
  },
  text: {
    fontSize: '1rem !important',
  },
  // same as for text templates -> generated text
  generatedTextValue: {
    borderRadius: '5px',
    minHeight: 'calc(1.5em + 0.75rem + 2px)',
    height: theme.typography.pxToRem(73),
    overflowY: 'auto',
    border: '1px solid #C4C4C4',
    padding: '0.375rem 0.75rem',
    marginBottom: '1rem',
    backgroundColor: RACCOLOR.CULTURED,
  },
}));

export interface WorkedHistoryNotesModalContentProps {
  workedResultText: string;
  noteText: string;
}
const WorkedHistoryNotesModalContent = ({
  workedResultText,
  noteText,
}: WorkedHistoryNotesModalContentProps) => {
  const classes = useStyles();
  return (
    <>
      <span>
        <Typography display="inline" variant="body1" className={classes.text}>
          Worked Result:{' '}
        </Typography>
        <Typography display="inline" variant="body2" className={classes.text}>
          {workedResultText}
        </Typography>
      </span>
      <Typography display="inline" variant="body2" className={classes.text}>
        Additional information on Activity Code
      </Typography>
      <span className={classes.generatedTextValue}>{noteText}</span>
    </>
  );
};

export const WorkedHistoryNotesModal = ({
  workedResultText,
  noteText,
  open,
  onClose,
}: ModalProps) => {
  const classes = useStyles();

  return (
    <RACModal
      isOpen={open}
      titleVariant="h4"
      classes={{
        dialogContent: classes.dialogContent,
        dialog: classes.dialogRoot,
        dialogActions: classes.dialogActions,
      }}
      maxWidth="sm"
      title=""
      content={
        <WorkedHistoryNotesModalContent
          workedResultText={workedResultText}
          noteText={noteText}
        />
      }
      onClose={onClose}
      buttons={
        <>
          <RACButton variant="contained" color="primary" onClick={onClose}>
            OK
          </RACButton>
        </>
      }
    />
  );
};
