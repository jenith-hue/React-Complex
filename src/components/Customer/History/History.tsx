import { makeStyles, RACCard, RACTabs } from '@rentacenter/racstrap';
import React from 'react';
import { AgreementInfoProvider } from '../../../context/AgreementInfo/AgreementInfoProvider';
import { AgreementInfo } from './AgreementInfo';
import { WorkedHistory } from './WorkedHistory';
import { CurrentPaymentHistory } from './CurrentPaymentHistory';
import { PaymentHistoryProvider } from '../../../context/PaymentHistory/PaymentHistoryProvider';

export const historyDatatestId = 'historyDatatestId';

const useClasses = makeStyles(() => ({
  historyCard: {
    boxSizing: 'border-box',
    borderRadius: '1rem',
    marginTop: '1rem',
    marginBottom: '9rem',
  },
  cardBody: {
    padding: '.5rem',
    width: '100%',
    boxSizing: 'border-box',
  },
}));

const AVAILABLE_TABS = [
  'Worked History',
  'Payment History',
  'Agreement Information',
];

export const History = () => {
  const classes = useClasses();

  return (
    <RACCard className={classes.historyCard} data-testid={historyDatatestId}>
      <div className={classes.cardBody}>
        <RACTabs
          loadAllTabContentOnMount
          defaultValue={0}
          tabs={AVAILABLE_TABS}
          contentForTabs={[
            <WorkedHistory key="worked-history" />,
            <PaymentHistoryProvider key="current-payment-history">
              <CurrentPaymentHistory />
            </PaymentHistoryProvider>,
            <AgreementInfoProvider key="agreement-info">
              <AgreementInfo />
            </AgreementInfoProvider>,
          ]}
        />
      </div>
    </RACCard>
  );
};
