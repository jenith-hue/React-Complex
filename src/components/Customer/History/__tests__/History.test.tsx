/* eslint-disable react/display-name */
/* eslint-disable sonarjs/no-identical-functions */

import React from 'react';
import renderer from 'react-test-renderer';
import { History } from '../History';

interface PropWithChildren {
  children: JSX.Element;
}

jest.mock('@rentacenter/racstrap', () => ({
  RACCard: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACCard" {...rest}>
      {children}
    </div>
  ),
  RACTabs: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="RACTabs" {...rest}>
      {children}
    </div>
  ),
  makeStyles: () => () => ({}),
}));

jest.mock('../WorkedHistory', () => ({
  WorkedHistory: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="WorkedHistory" {...rest}>
      {children}
    </div>
  ),
}));

jest.mock('../AgreementInfo', () => ({
  AgreementInfo: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="AgreementInfo" {...rest}>
      {children}
    </div>
  ),
}));

jest.mock('../CurrentPaymentHistory', () => ({
  CurrentPaymentHistory: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="CurrentPaymentHistory" {...rest}>
      {children}
    </div>
  ),
}));

jest.mock('../../../../context/AgreementInfo/AgreementInfoProvider', () => ({
  AgreementInfoProvider: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="AgreementInfoProvider" {...rest}>
      {children}
    </div>
  ),
}));

jest.mock('../../../../context/PaymentHistory/PaymentHistoryProvider', () => ({
  PaymentHistoryProvider: ({ children, ...rest }: PropWithChildren) => (
    <div data-testid="PaymentHistoryProvider" {...rest}>
      {children}
    </div>
  ),
}));

describe('History', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<History />);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
