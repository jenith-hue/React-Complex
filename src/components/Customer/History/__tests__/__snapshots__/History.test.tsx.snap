// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History renders correctly 1`] = `
<div
  data-testid="historyDatatestId"
>
  <div>
    <div
      contentForTabs={
        Array [
          <WorkedHistory />,
          <PaymentHistoryProvider>
            <CurrentPaymentHistory />
          </PaymentHistoryProvider>,
          <AgreementInfoProvider>
            <AgreementInfo />
          </AgreementInfoProvider>,
        ]
      }
      data-testid="RACTabs"
      defaultValue={0}
      loadAllTabContentOnMount={true}
      tabs={
        Array [
          "Worked History",
          "Payment History",
          "Agreement Information",
        ]
      }
    />
  </div>
</div>
`;
